pybaseball/playerid_lookup.py

```diff
@@ -84,6 +99,24 @@ def playerid_lookup(last=None, first=None, player_list=None):
 
     results = results.reset_index(drop=True)
 
+    if len(results) == 0:
+        similarity_df = name_similarity(last=last, first=first, player_table=table)
+        minimum_distance = similarity_df[similarity_df["total_dist"] == similarity_df["total_dist"].min()]
+
+        # If there's only one similar name, return that
+        if len(minimum_distance) == 1:
+            suggested_first = minimum_distance["name_first"].iloc[0].capitalize()
+            suggested_last = minimum_distance["name_last"].iloc[0].capitalize()
+            raise ValueError(f"Player not found! Did you mean {suggested_first} {suggested_last}?")
```

Comment:
I'm not sure if raising a value error is the best thing here. I'd prefer returning an empty result. What's the rationale to raise an error?

Reply:
I prefer the notification when the query fails vs silent failure. If the preference is for no errors, fine, but then the question is what the preferred method of return is? I could see:

1) Print guesses to screen and also return the empty DataFrame
2) Show the nearest X players to the DataFrame with a print statement of "no identical matches found, closest X returned"

Reply:
I think we;re mixing up two concepts here, lookup vs search. thinking it through now, I'd expect a lookup with a missing key to raise an error and a search that has no results to return an empty result. so I don;t know where that leaves us - any other thoughts?

Reply:
Probably we should think of the use-case. Personally, when I use this function, I'm starting up an analysis and just trying to find keys for a player I want to study. This means I'm typically working in a Jupyter notebook. In that case, errors are ugly and generally not useful, imo. However, an empty data frame _also_ isn't useful (also it's super frustrating because this function takes some time to get the data, and I know I have to run it again).

If that's the main and/or only use-case, I think solution 2 above works best, showing the X closest players to the entered name with some warning print statement. I just finished writing this code (w/fuzzywuzzy) so that's workable.

However, if you can think of use-cases in which having the empty dataframe or an error message is better for the user, we should probably default to that. None immediately come to mind, but ofc my scope is limited to my experience.

Reply:
>However, if you can think of use-cases in which having the empty dataframe or an error message is better for the user, we should probably default to that. None immediately come to mind, but ofc my scope is limited to my experience.

I just thought some more about this - I could imagine someone doing something conditional on the size of the returned df. Maybe the best option is to add an optional boolean argument `search`, when True it will return the 5 nearest suggestions? We could default it to False so it doesn't disrupt current uses. The only con I see with doing that approach is that I don't imagine many people will use it in that case, but oh well.